google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -94,21 +83,18 @@
   private final String cachedTopicNameString;
 
   private final BatchingSettings batchingSettings;
-  private final RetrySettings retrySettings;
-  private final LongRandom longRandom;
 
   private final Lock messagesBatchLock;
   private List<OutstandingPublish> messagesBatch;
   private int batchedBytes;
 
   private final AtomicBoolean activeAlarm;
 
-  private final Channel channel;
-  @Nullable private final CallCredentials callCredentials;
+  private final TopicAdminClient topicClient;
```

Comment:
Could we directly use `PublisherStub` instead of `TopicAdminClient`? It is conceptually cleaner, and then we can actually completely remove the `publish` call from `TopicAdminClient` instead of having it package-private.

Reply:
Done.